components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/ClaimDialectDAO.java

```diff
@@ -136,4 +136,30 @@ public void removeClaimDialect(ClaimDialect claimDialect, int tenantId) throws C
             IdentityDatabaseUtil.closeConnection(connection);
         }
     }
+
+    /**
+     * Remove all claim dialects of a given tenant.
+     *
+     * @param tenantId Id of the tenant
+     * @throws ClaimMetadataException
+     */
+    public void removeAllClaimDialects(int tenantId) throws ClaimMetadataException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement prepStmt = null;
+
+        String query = SQLConstants.REMOVE_CLAIM_DIALECTS_BY_TENANT_ID;
+        try {
```

Comment:
Any reason here to not to use try with resource option?

Reply:
Fixed in #3312